### PR TITLE
feat: Improving test coverage of SCs

### PR DESCRIPTION
**Pull request by Raid5594**
# What ❔

Add tests for contracts or simply factor existing tests, so that they are included in coverage report.

## Why ❔

Increase overall coverage of smart contracts in the repo

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l1-contracts/contracts/dev-contracts/test/AddressAliasHelperTest.sol [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import "../../vendor/AddressAliasHelper.sol";
+
+contract AddressAliasHelperTest {
```
Comment (vladbochok) [inline l1-contracts/contracts/dev-contracts/test/AddressAliasHelperTest.sol:7]: Without wrapper test coverage doesn't work?

File: l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol [hunk 2]
```diff
@@ -5,38 +5,33 @@ pragma solidity 0.8.20;
...
     PriorityQueue.Queue priorityQueue;
 
     function getFirstUnprocessedPriorityTx() external view returns (uint256) {
-        return priorityQueue.getFirstUnprocessedPriorityTx();
+        return PriorityQueue.getFirstUnprocessedPriorityTx(priorityQueue);
     }
 
     function getTotalPriorityTxs() external view returns (uint256) {
-        return priorityQueue.getTotalPriorityTxs();
+        return PriorityQueue.getTotalPriorityTxs(priorityQueue);
```
Comment (vladbochok) [inline l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol:15]: Why?

File: l1-contracts/test/foundry/unit/concrete/AddressAliasHelper/_AddressAliasHelper_Shared.t.sol [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {Test} from "forge-std/Test.sol";
+import {AddressAliasHelperTest} from "solpp/dev-contracts/test/AddressAliasHelperTest.sol";
+
+contract AddressAliasHelperSharedTest is Test {
```
Comment (vladbochok) [inline l1-contracts/test/foundry/unit/concrete/AddressAliasHelper/_AddressAliasHelper_Shared.t.sol:7]: Please add `test()` function to remove this file from the coverage?

**Review by vladbochok — COMMENTED** [review]
Please make sure no `dev-contracts`/`tests`/`artifacts` contracts in `yarn coverage:foundry`

File: l1-contracts/contracts/dev-contracts/test/AddressAliasHelperTest.sol (hunk 1 above)
Comment (Raid5594) [reply l1-contracts/contracts/dev-contracts/test/AddressAliasHelperTest.sol:7]: No, the coverage ignores the library if it is used directly

File: l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol (hunk 2 above)
Comment (Raid5594) [reply l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol:15]: https://github.com/foundry-rs/foundry/pull/3128

`using for ` patterns are ignored by coverage

File: l1-contracts/test/foundry/unit/concrete/AddressAliasHelper/_AddressAliasHelper_Shared.t.sol (hunk 3 above)
Comment (Raid5594) [reply l1-contracts/test/foundry/unit/concrete/AddressAliasHelper/_AddressAliasHelper_Shared.t.sol:7]: done

File: l1-contracts/test/foundry/unit/concrete/common/libraries/UncheckedMath/_UncheckedMath_Shared.t.sol [hunk 4]
```diff
@@ -2,5 +2,12 @@
 pragma solidity 0.8.20;
 
 import {Test} from "forge-std/Test.sol";
+import {UncheckedMathTest} from "solpp/dev-contracts/test/UncheckedMathTest.sol";
 
-contract UncheckedMathTest is Test {}
+contract UncheckedMathSharedTest is Test {
```
[the remark this answers is not in the document]
Comment (Raid5594) [reply l1-contracts/test/foundry/unit/concrete/common/libraries/UncheckedMath/_UncheckedMath_Shared.t.sol:7]: done

File: l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol (hunk 2 above)
Comment (vladbochok) [reply l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol:18]: You should add back, thanks!

File: l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol [hunk 5]
```diff
@@ -8,35 +8,33 @@ contract PriorityQueueTest {
     // add this to be excluded from coverage report
     function test() internal virtual {}
```
Comment (vladbochok) [inline l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol:9]: This should be removed in order to `PriorityQueue.sol` to be in coverage

File: l1-contracts/contracts/dev-contracts/test/UncheckedMathTest.sol [hunk 6]
```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import "../../common/libraries/UncheckedMath.sol";
+
+contract UncheckedMathTest {
+    // add this to be excluded from coverage report
+    function test() internal virtual {}
```
Comment (vladbochok) [inline l1-contracts/contracts/dev-contracts/test/UncheckedMathTest.sol:9]: Same here

File: l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol (hunk 2 above)
Comment (Raid5594) [reply l1-contracts/contracts/dev-contracts/test/PriorityQueueTest.sol:15]: It moved to state-transition folder, so all good, missed it :)

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    function testRevertWhenCalledNotByBridgehub() public {
+        address baseToken = address(0x3030303);
+        address sharedBridge = address(0x4040404);
+        address admin = bridgehub;
+        uint256 chainId = block.chainid;
+        Diamond.DiamondCutData memory initialDiamondCutData = getDiamondCutData(diamondInit);
+
+        vm.expectRevert(bytes("StateTransition: only bridgehub"));
```
Comment (Raid5594) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol:30]: It doesn't seem that coverage captured modifier's revert, researching how to make it work

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+        chainContractAddress.createNewChain(chainId, baseToken, sharedBridge, admin, abi.encode(initialDiamondCutData));
```
Comment (Raid5594) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol:32]: Will probably move new chain creation to a function, as it's been reused in multiple places

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/_StateTransitionManager_Shared.t.sol [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+            admin,
+            abi.encodeCall(StateTransitionManager.initialize, stmInitializeData)
+        );
+        chainContractAddress = StateTransitionManager(address(transparentUpgradeableProxy));
+
+        vm.stopPrank();
+        vm.startPrank(governor);
+    }
+
+    function getAdminSelectors() private view returns (bytes4[] memory) {
```
Comment (Raid5594) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/_StateTransitionManager_Shared.t.sol:115]: Will move selector getters to Utils

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/FreezeChain.t.sol [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+        chainContractAddress.freezeChain(block.chainid);
+
+        // Repeated call should revert
+        vm.expectRevert(bytes.concat("q1")); // storage frozen
+        chainContractAddress.freezeChain(block.chainid);
+
+        // Call fails as storage is frozen
+        vm.expectRevert(bytes.concat("q1"));
+        isChainFrozen = gettersFacet.isDiamondStorageFrozen();
```
Comment (Raid5594) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/FreezeChain.t.sol:43]: Is it the case that this getter can't return true? (as the call reverts with the error described above)

File: l1-contracts/test/foundry/unit/concrete/Utils/Utils.sol [hunk 11]
```diff
@@ -92,6 +92,58 @@ library Utils {
...
+        );
+        logs[7] = constructL2Log(true, PUBDATA_PUBLISHER_ADDRESS, uint256(SystemLogKey.BLOB_ONE_HASH_KEY), bytes32(0));
+        logs[8] = constructL2Log(true, PUBDATA_PUBLISHER_ADDRESS, uint256(SystemLogKey.BLOB_TWO_HASH_KEY), bytes32(0));
+        logs[9] = constructL2Log(
+            true,
+            L2_BOOTLOADER_ADDRESS,
+            uint256(SystemLogKey.EXPECTED_SYSTEM_CONTRACT_UPGRADE_TX_HASH_KEY),
+            _expectedSystemContractUpgradeTxHash
+        );
+        return logs;
```
[the remark this answers is not in the document]
Comment (Raid5594) [reply l1-contracts/test/foundry/unit/concrete/Utils/Utils.sol:144]: done

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/_StateTransitionManager_Shared.t.sol (hunk 9 above)
Comment (Raid5594) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/_StateTransitionManager_Shared.t.sol:115]: done

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol (hunk 7 above)
Comment (Raid5594) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol:30]: done

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol (hunk 8 above)
Comment (Raid5594) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol:32]: done

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol [hunk 12]
```diff
@@ -0,0 +1,58 @@
+// // SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {StateTransitionManagerTest} from "./_StateTransitionManager_Shared.t.sol";
+import {Diamond} from "solpp/state-transition/libraries/Diamond.sol";
+import {DiamondProxy} from "solpp/state-transition/chain-deps/DiamondProxy.sol";
+
+contract createNewChainTest is StateTransitionManagerTest {
+    function testRevertWhenInitialDiamondCutHashMismatch() public {
```
[the remark this answers is not in the document]
Comment (Raid5594) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol]: done

**Review by vladbochok — APPROVED** [review]

**Pull request merged**